homeassistant/components/airq/config_flow.py

```diff
@@ -0,0 +1,123 @@
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_IP_ADDRESS): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(data: dict[str, Any]) -> None:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    try:
+        auth_success = await AirQ(
+            data[CONF_IP_ADDRESS], data[CONF_PASSWORD]
+        ).test_authentication()
+    except ClientConnectionError as exc:
+        raise CannotConnect from exc
+
+    if not auth_success:
+        raise InvalidAuth
+
+
+async def fetch_device_info(data: dict[str, Any]) -> tuple[str, str]:
+    """Fetch device information: name and a unique ID."""
+    airq = AirQ(data[CONF_IP_ADDRESS], data[CONF_PASSWORD])
+    config = await airq.get("config")
+    device_id: str = config["id"]
+    device_name: str = config["devicename"]
+    return device_name, device_id
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for air-Q."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial configuration step: authentication.
+
+        This prompts the user to enter the credentials to access the device.
+        It then tries to connect to the device, and to assess the validity
+        of the password.
+
+        If authentication is successful, next step handles minimal configuration
+        for the integration.
+        """
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            await validate_input(user_input)
+        except CannotConnect:
+            # Is it ok to print the IP? Looks so, judging by this
+            # https://developers.home-assistant.io/docs/development_guidelines?_highlight=print&_highlight=out#log-messages
+            # Also, I am opting for .error over .exception since the latter includes
+            # a humongous traceback, which looks far too scary for such a minor thing
+            _LOGGER.error(
+                "Failed to connect to device %s. Check the specified IP address / mDNS, "
+                "as well as whether the device is connected to power and the WiFi",
+                user_input[CONF_IP_ADDRESS],
+            )
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            _LOGGER.error(
+                "Incorrect password for device %s", user_input[CONF_IP_ADDRESS]
+            )
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            # This really shouldn't happen, so .exception is perhaps more appropriate
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
```

Comment:
You shouldn't have to do this, the base config flow handler should handle it and give a reasonable error message to the user already.
Try raising some arbitrary exception and see what happens.

Reply:
Perhaps, I am missing something here. Simply omitting this except-clause fails `tests.components.airq.test_config_flow.test_form_unknown_error`. Am I testing a wrong thing there?